gemma-web/src/main/java/ubic/gemma/web/controller/OntologyController.java

```diff
@@ -67,14 +75,36 @@ public RedirectView getTerm( @PathVariable("termId") String termId ) {
         if ( term == null ) {
             throw new EntityNotFoundException( String.format( "No term with IRI %s in TGEMO.", iri ) );
         }
-        String gemmaOntologyUrl = gemmaOntologyService.getOntologyUrl();
-        String urlWithFragment = UriComponentsBuilder.fromHttpUrl( gemmaOntologyUrl )
-                .fragment( iri )
-                .build()
-                .toUriString();
-        RedirectView redirectView = new RedirectView( urlWithFragment );
-        redirectView.setStatusCode( HttpStatus.FOUND );
-        return redirectView;
+
+        StringBuilder s = new StringBuilder( String.format( "<title>%s</title>", escapeHtml4( term.getLabel() ) ) +
+                "<div class=\"padded\"><ul style=\"padding-bottom:0.5em\">" +
+                String.format( "<h2 style=\"padding-bottom:0.5em\">%s: %s</h2>",
+                        escapeHtml4( term.getLocalName() ),
+                        escapeHtml4( term.getLabel() ) ));
+                if ( term.getUri() != null) {
+                    s.append(String.format( "<li><a href=%s>%s</a></li>",
+                            escapeHtml4(term.getUri()
+                                    .replaceFirst( "^" + Pattern.quote( TGEMO_URI_PREFIX ), servletContext.getContextPath() + "/ont/" ) ) ,
+                            escapeHtml4( TGEMO_URI_PREFIX + term.getLocalName() ) ) );
```

Comment:
The local name by definition does not include a URI prefix, unless the term if from a foreign ontology.